internal/handlers/sqlite/msg_dropindexes.go

```diff
@@ -16,13 +16,234 @@ package sqlite
+	}
+
+	db, err := h.b.Database(dbName)
+	if err != nil {
+		if backends.ErrorCodeIs(err, backends.ErrorCodeDatabaseNameIsInvalid) {
+			msg := fmt.Sprintf("Invalid namespace specified '%s.%s'", dbName, collection)
+			return nil, commonerrors.NewCommandErrorMsgWithArgument(commonerrors.ErrInvalidNamespace, msg, command)
+		}
+
+		return nil, lazyerrors.Error(err)
+	}
+
+	c, err := db.Collection(collection)
+	if err != nil {
+		if backends.ErrorCodeIs(err, backends.ErrorCodeCollectionNameIsInvalid) {
+			msg := fmt.Sprintf("Invalid namespace specified '%s.%s'", dbName, collection)
+			return nil, commonerrors.NewCommandErrorMsgWithArgument(commonerrors.ErrInvalidNamespace, msg, command)
+		}
+
+		return nil, lazyerrors.Error(err)
+	}
+
+	indexValue, err := document.Get("index")
+	if err != nil {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrMissingField,
+			"BSON field 'dropIndexes.index' is missing but a required field",
+			command,
+		)
+	}
+
+	beforeDrop, err := c.ListIndexes(ctx, nil)
+	if err != nil {
+		if backends.ErrorCodeIs(err, backends.ErrorCodeCollectionDoesNotExist) {
+			msg := fmt.Sprintf("ns not found %s.%s", dbName, collection)
+			return nil, commonerrors.NewCommandErrorMsgWithArgument(commonerrors.ErrNamespaceNotFound, msg, command)
+		}
+
+		return nil, lazyerrors.Error(err)
+	}
+
+	toDrop, dropAll, err := processDropIndexOptions(command, indexValue, beforeDrop.Indexes)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = c.DropIndexes(ctx, &backends.DropIndexesParams{Indexes: toDrop})
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	replyDoc := must.NotFail(types.NewDocument(
+		"nIndexesWas", int32(len(beforeDrop.Indexes)),
+	))
+
+	if dropAll {
+		replyDoc.Set("msg", "non-_id indexes dropped for collection")
+	}
+
+	replyDoc.Set(
+		"ok", float64(1),
+	)
+
+	var reply wire.OpMsg
+	must.NoError(reply.SetSections(wire.OpMsgSection{
+		Documents: []*types.Document{replyDoc},
+	}))
+
+	return &reply, nil
+}
+
+// processDropIndexOptions parses index doc and returns the list of indexes to delete
+// and true if a flag to drop all indexes except _id_ was set.
+func processDropIndexOptions(command string, v any, existing []backends.IndexInfo) ([]string, bool, error) { //nolint:lll // for readability
+	switch v := v.(type) {
+	case *types.Document:
+		// Index specification (key) is provided to drop a specific index.
+		var spec []backends.IndexKeyPair
+
+		spec, err := processIndexKey(command, v)
+		if err != nil {
+			return nil, false, lazyerrors.Error(err)
+		}
+
+		if len(spec) == 1 && spec[0].Field == "_id" && !spec[0].Descending {
+			return nil, false, commonerrors.NewCommandErrorMsgWithArgument(
+				commonerrors.ErrInvalidOptions, "cannot drop _id index", command,
+			)
+		}
+
+		for _, index := range existing {
+			for i, key := range index.Key {
+				if key.Field == spec[i].Field && key.Descending == spec[i].Descending {
+					return []string{index.Name}, false, nil
```

Comment:
We are returning if first field matches with `spec[i].Field`. Don't we want to check if all `Field` match? :thinking: 

For example, if index contains two fields, it will match as long as first field is the same.

Reply:
Nice catch! Our tests didn't find it, so I think I should add a test where the first field matches and the second doesn't.